src/components/utils/src/file_system_qt.cc

```diff
@@ -0,0 +1,357 @@
+    return qFileInfo.isWritable();
+   case(R_OK):
+    return qFileInfo.isReadable();
+   default:
+       return false;
+   }
+}
+
+bool file_system::IsWritingAllowed(const std::string& name) {
+  return IsAccessible(name, W_OK);
+}
+
+bool file_system::IsReadingAllowed(const std::string& name) {
+  return IsAccessible(name, R_OK);
+}
+
+std::vector<std::string> file_system::ListFiles(
+  const std::string& directory_name) {
+  std::vector<std::string> listFiles;
+  if (!DirectoryExists(directory_name)) {
+    return listFiles;
+  }
+  QDir dir(QString(directory_name.c_str()));
+  QStringList list_files = dir.entryList(QDir::Files);
+  foreach( QString str, list_files) {
+    listFiles.push_back(str.toStdString());
+  }
+  return listFiles;
+}
+
+bool file_system::WriteBinaryFile(const std::string& name,
+                                  const std::vector<uint8_t>& contents) {
+  using namespace std;
+  ofstream output(name.c_str(), ios_base::binary|ios_base::trunc);
+  output.write(reinterpret_cast<const char*>(&contents.front()),
+               contents.size());
+  return output.good();
+}
+
+bool file_system::ReadBinaryFile(const std::string& name,
+                                 std::vector<uint8_t>& result) {
+  if (!FileExists(name) || !IsAccessible(name, R_OK)) {
+    return false;
+  }
+
+  std::ifstream file(name.c_str(), std::ios_base::binary);
+  std::ostringstream ss;
+  ss << file.rdbuf();
+  const std::string& s = ss.str();
+
+  result.resize(s.length());
+  std::copy(s.begin(), s.end(), result.begin());
+  return true;
+}
+
+bool file_system::ReadFile(const std::string& name, std::string& result) {
+  if (!FileExists(name) || !IsAccessible(name, R_OK)) {
+    return false;
+  }
+
+  std::ifstream file(name.c_str());
+  std::ostringstream ss;
+  ss << file.rdbuf();
+  result = ss.str();
+  return true;
+}
+
+const std::string file_system::ConvertPathForURL(const std::string& path) {
+  QString p(path.c_str());
+  return QUrl::fromLocalFile(p).url().toStdString();
+}
+
+bool file_system::CreateFile(const std::string& path) {
+  std::ofstream file(path);
+  if (!(file.is_open())) {
+    return false;
+  } else {
+    file.close();
+    return true;
+  }
+}
+
+
+uint64_t file_system::GetFileModificationTime(const std::string& path) {
+    QFileInfo f(QString(path.c_str()));
+    return static_cast<uint64_t>(f.lastModified().toMSecsSinceEpoch());
+}
+
+bool file_system::CopyFile(const std::string& src,
+                           const std::string& dst) {
+  if (!FileExists(src) || FileExists(dst) || !CreateFile(dst)) {
+    return false;
+  }
+    return QFile::copy(QString(src.c_str()), QString(dst.c_str()));
+}
+
+bool file_system::MoveFile(const std::string& src,
+                           const std::string& dst) {
+  if (!CopyFile(src, dst)) {
+    return false;
+  }
+  if (!DeleteFile(src)) {
+    DeleteFile(dst);
+    return false;
+  }
+  return true;
+}
+
+std::string file_system::GetPathDelimiter() {
+  return "\\";
```

Comment:
We have to use Qt native and cross-platform solution. Possibly it worth to return just "\" on windows and update client code.